interpreter/src/main/scala/pl/touk/nussknacker/engine/Interpreter.scala

```diff
@@ -68,7 +68,7 @@ private class InterpreterInternal[F[_]](listeners: Seq[ProcessListener],
         interpretNext(next, newCtx.pushNewContext(vars))
       case SubprocessEnd(_, varName, fields, next) =>
         val updatedCtx = createOrUpdateVariable(ctx, varName, fields)
-        val parentContext = ctx.popContext
+        val parentContext = ctx.parentContext.getOrElse(updatedCtx.copy(variables = Map.empty))
```

Comment:
can we add some comment/todo why we reuse this context instead of creating a new one?

Reply:
done